src/main/java/com/google/devtools/build/lib/remote/RemoteModule.java

```diff
@@ -160,6 +172,13 @@ public void beforeCommand(CommandEnvironment env) throws AbruptExitException {
     boolean enableHttpCache = RemoteCacheClientFactory.isHttpCache(remoteOptions);
     boolean enableGrpcCache = GrpcCacheClient.isRemoteCacheOptions(remoteOptions);
     boolean enableRemoteExecution = shouldEnableRemoteExecution(remoteOptions);
+    boolean enableRemoteDownloader = shouldEnableRemoteDownloader(remoteOptions);
+
+    if (enableRemoteDownloader && !enableGrpcCache) {
```

Comment:
This seems to require `--remote_cache` be set alongside `--remote_executor` are there any side effects which could arise from changing this to the below snippet?

```java
if (enableRemoteDownloader && (!enableGrpcCache || !enableRemoteExecution)) {
```

Reply:
Remote execution is not required to use remote downloads. Remote caching (specifically gRPC remote caching) _is_ required, because that's where the downloads get placed.

Reply:
I've opened https://github.com/bazelbuild/bazel/issues/11913 to track this issue.